fre/app/generate_time_averages/frenctoolsTimeAverager.py

```diff
@@ -17,7 +17,17 @@ class inheriting from abstract base class timeAverager
     '''
 
     def generate_timavg(self, infile=None, outfile=None):
-        ''' use fre-nctool's CLI timavg.csh with subprocess call '''
+        """
+        use fre-nctool's CLI timavg.csh with subprocess call
+
+        :param self: 
```

Comment:
I think the recommendation will be to document this as `:param self: This is an instance of the class frenctoolsTimeAverager` and I would not define a type for it.

So you can keep self as the first parameter and document it here, and then delete lines 28 and 29 I think.